src/Features/Core/Portable/Copilot/IProposalAdjusterService.cs

```diff
@@ -46,10 +48,24 @@ ValueTask<ProposalAdjustmentResult> TryAdjustProposalAsync(
         Checksum solutionChecksum, DocumentId documentId, ImmutableArray<TextChange> normalizedChanges, CancellationToken cancellationToken);
 }
 
-internal abstract class AbstractCopilotProposalAdjusterService() : ICopilotProposalAdjusterService
+internal abstract class AbstractCopilotProposalAdjusterService : ICopilotProposalAdjusterService
 {
-    protected abstract Task<ImmutableArray<TextChange>> AddMissingTokensIfAppropriateAsync(
-        Document originalDocument, ImmutableArray<TextChange> normalizedChanges, CancellationToken cancellationToken);
+    protected readonly IGlobalOptionService globalOptions;
+
+    private readonly ImmutableArray<(string name, Adjuster adjuster)> _adjusters;
```

Comment:
note: if the data gets more complex here, we likely want to move to an interface to encapsulate it.